fakestorage/object.go

```diff
@@ -516,7 +516,7 @@ func (s *Server) downloadObject(w http.ResponseWriter, r *http.Request) {
 
 	status := http.StatusOK
 	start, end, content := s.handleRange(obj, r)
-	if len(content) != len(obj.Content) {
+	if r.Header.Get("Range") != "" {
```

Comment:
Let's have `handleRange` return whether or not this is a "ranged" request, this way only that method is dealing with request headers for range.

Reply:
Done.